### PR TITLE
s3ql 5.1.2

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Upstream seems active again, so let's undeprecated. Will probably need some more updates/bumps until it works, let's see..

**Comment by chenrui333** [comment]
good catch! 👍

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
```
  ==> /home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/bin/mount_s3ql --version
  Traceback (most recent call last):
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/bin/mount_s3ql", line 5, in <module>
      from s3ql.mount import main
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/mount.py", line 31, in <module>
      from . import fs
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/fs.py", line 24, in <module>
      from .backends.common import CorruptedObjectError, NoSuchObject
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/backends/__init__.py", line 9, in <module>
      from . import gs, local, rackspace, s3, s3c, swift, swiftks
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/backends/rackspace.py", line 13, in <module>
      from . import swiftks
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/backends/swiftks.py", line 18, in <module>
      from . import swift
    File "/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/lib/python3.9/site-packages/s3ql/backends/swift.py", line 26, in <module>
      from packaging.version import Version
  ModuleNotFoundError: No module named 'packaging'
```

File: Formula/s/s3ql.rb [hunk 1]
```diff
@@ -3,29 +3,27 @@ class S3ql < Formula
 
   desc "POSIX-compliant FUSE filesystem using object store as block storage"
   homepage "https://github.com/s3ql/s3ql"
-  url "https://github.com/s3ql/s3ql/releases/download/release-3.8.1/s3ql-3.8.1.tar.gz"
-  sha256 "d4731ebaacadca38a677bb18a99446c19d4f5b573628d55371f715acace11c4c"
-  license "GPL-3.0"
-  revision 1
+  url "https://github.com/s3ql/s3ql/releases/download/s3ql-5.1.2/s3ql-5.1.2.tar.gz"
+  sha256 "5921f6286247792ecb93cd78105eaa5d71e8b3037628ae1bce5c78dc6cf08fd2"
+  license "GPL-3.0-or-later"
```
Comment (chenrui333) [inline Formula/s/s3ql.rb:8]: I think upstream used a deprecated license.

Comment (SMillerDev) [reply Formula/s/s3ql.rb:8]: Source doesn't include the "or any later version" in the file headers. It should be this:
```suggestion
  license "GPL-3.0-only"
```

**Comment by iMichka** [comment]
Does not build with Python 3.12:

```
  gcc-11 -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I/home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/libexec/include -I/home/linuxbrew/.linuxbrew/opt/python@3.12/include/python3.12 -c src/s3ql/sqlite3ext.cpp -o build/temp.linux-x86_64-cpython-312/src/s3ql/sqlite3ext.o -Wall -Wextra -Wconversion -Wsign-compare
  src/s3ql/sqlite3ext.cpp: In function ‘void __Pyx_Raise(PyObject*, PyObject*, PyObject*, PyObject*)’:
  src/s3ql/sqlite3ext.cpp:3636:36: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
   3636 |         PyObject* tmp_tb = tstate->curexc_traceback;
        |                                    ^~~~~~~~~~~~~~~~
  src/s3ql/sqlite3ext.cpp:3639:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘curexc_traceback’
   3639 |             tstate->curexc_traceback = tb;
        |                     ^~~~~~~~~~~~~~~~
  src/s3ql/sqlite3ext.cpp: In function ‘size_t __Pyx_PyInt_As_size_t(PyObject*)’:
  src/s3ql/sqlite3ext.cpp:4413:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
   4413 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
        |                                                       ^~~~~~~~
  src/s3ql/sqlite3ext.cpp:4468:55: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
```

File: Formula/s/s3ql.rb (hunk 1 above)
Comment (iMichka) [reply Formula/s/s3ql.rb]: Ok right, it's fixed now

**Comment by chenrui333** [comment]
```
  ==> /home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/bin/mount_s3ql --version
  ==> ls
  ==> /home/linuxbrew/.linuxbrew/Cellar/s3ql/5.1.2/bin/mkfs_s3ql --plain local:///tmp/s3ql-test-20231030-15320-14k72yp 2>&1
  ==> ls s3ql_metadata
  ls: cannot access 's3ql_metadata': No such file or directory
  
  Error: s3ql: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected: 0
    Actual: 2
```

**Comment by iMichka** [comment]
Should be good now, it works :)

**Review by singingwolfboy — APPROVED** [review]

**Pull request merged**